pkg/config/api.go

```diff
@@ -0,0 +1,98 @@
+package config
+
+import (
+	"strings"
+
+	"github.com/supabase/cli/internal/utils/cast"
+	"github.com/supabase/cli/internal/utils/diff"
+	v1API "github.com/supabase/cli/pkg/api"
+)
+
+type (
+	RemoteApi struct {
+		Enabled         bool     `toml:"enabled"`
+		Schemas         []string `toml:"schemas"`
+		ExtraSearchPath []string `toml:"extra_search_path"`
+		MaxRows         uint     `toml:"max_rows"`
+	}
+	api struct {
+		RemoteApi
+		Image     string  `toml:"-"`
+		KongImage string  `toml:"-"`
+		Port      uint16  `toml:"port"`
+		Tls       tlsKong `toml:"tls"`
+		// TODO: replace [auth|studio].api_url
+		ExternalUrl string `toml:"external_url"`
+	}
+
+	tlsKong struct {
+		Enabled bool `toml:"enabled"`
+	}
+)
+
+func (a *RemoteApi) ToUpdatePostgrestConfigBody() v1API.UpdatePostgrestConfigBody {
+	body := v1API.UpdatePostgrestConfigBody{}
+
+	// When the api is disabled, remote side it just set the dbSchema to an empty value
+	if !a.Enabled {
+		emptyString := ""
+		body.DbSchema = &emptyString
+		return body
+	}
+
+	// Convert Schemas to a comma-separated string
+	if len(a.Schemas) > 0 {
+		schemas := strings.Join(a.Schemas, ",")
+		body.DbSchema = &schemas
+	}
+
+	// Convert ExtraSearchPath to a comma-separated string
+	if len(a.ExtraSearchPath) > 0 {
+		extraSearchPath := strings.Join(a.ExtraSearchPath, ",")
+		body.DbExtraSearchPath = &extraSearchPath
+	}
+
+	// Convert MaxRows to int pointer
+	if a.MaxRows > 0 {
+		intValue := cast.UintToInt(a.MaxRows)
+		body.MaxRows = &intValue
+	}
+
+	// Note: DbPool is not present in the Api struct, so it's not set here
+	return body
+}
+
+func (a *RemoteApi) fromRemoteApiConfig(remoteConfig v1API.PostgrestConfigWithJWTSecretResponse) RemoteApi {
```

Comment:
```suggestion
func NewRemoteApiConfig(remoteConfig v1API.PostgrestConfigWithJWTSecretResponse) remoteApi {
```

Do we need to override existing config? If not, it's more idiomatic to declare the constructor on its own.

Reply:
I do actually "override" the existing config. We don't replace it since we clone it but we keep it's fields and "replace" the ones coming from the "remoteConfig". That allow for having a cleaner diff in some cases (for instance, when Enabled = false), since the rest of the "RemoteConfig" will be kept the same. Rather than having a diff with empty db schemas.